### DISCUSSION TITLE
Complete github context: best way to see it?

### DISCUSSION THREAD

**Discussion by mathomp4** [community/community#25624] [Actions]
<p>All,</p>
<p>I know that the <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="noopener nofollow ugc"><code>github</code> context</a> is quite powerful and so I’m wondering if the complete “API” exists anywhere? (For example, from that page you wouldn’t know <code>github.event.repository.name</code> exists because it only talks up to <code>github.event</code>.)</p>
<p>I did see the bits in the doc about <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#example-printing-context-information-to-the-log-file" rel="noopener nofollow ugc"><code>toJSON</code>'ing the context in a workflow</a>, but the warning on that link:</p>
<blockquote>
<p><strong>Warning:</strong> When using the whole <code>github</code> context, be mindful that it includes sensitive information such as <code>github.token</code>. GitHub masks secrets when they are printed to the console, but you should be cautious when exporting or printing the context.</p>
</blockquote>
<p>sort of frightens me as someone who once accidentally exposed a key on GitHub and found out the speed and power of hackers monitoring GitHub! Is this a fairly safe way to see the full context? Or maybe there’s a way to use <a href="https://github.com/mxschmitt/action-tmate" rel="noopener nofollow ugc"><code>action-tmate</code></a> to do it in a safe way?</p>

**Comment by jsoref** [comment, marked as answer]
<p>If your repository is private to only you, the <code>toJSON</code> should be reasonably safe.</p>
<p>I wouldn’t do it otherwise if I were you. – Having had a white hat actually find a valid exploit against one of my workflows.</p>

**Comment by mathomp4** [comment]
<p>Yeah. I figured. I did this on a throwaway and I got what I needed for my task!</p>
